ls1intum/Artemis
# `Development`: Persist checkout solution repository for programming exercises with information from Bamboo

### Checklist
#### General


#### Server

### Motivation and Context


In preparation of migrating away from Bamboo, we need to persist information that resides solely in Bamboo. 
### Description
This PR checks if the solution repository is checked out in the BASE Build Plan and stores that information in the database.
### Steps for Testing

Prerequisites:
- 1 Atlassian Setup with Haskell and Ocaml PEs

1. Start Artemis
2. Check in the database and bamboo if the values match


For admins, the DB change is reversible with:
`alter table programming_exercise_details drop column checkout_solution_repository;`
### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
